sphinx/builders/gettext.py

```diff
@@ -32,13 +32,14 @@ class Message:
     """An entry of translatable message."""
     def __init__(self, text: str, locations: List[Tuple[str, int]], uuids: List[str]):
         self.text = text
-        self.locations = locations
+        self.locations = list(sorted(list(set(locations))))     #make list unique and sorted alphabetically
```

Comment:
`sorted` returns a list and can take a set as input

```suggestion
        self.locations = sorted(set(locations))     #make list unique and sorted alphabetically
```